source/agora/consensus/validation/Enrollment.d

```diff
@@ -93,6 +95,9 @@ public string isInvalidReason (in Enrollment enrollment,
         return Message;
     }
 
+    if (hasPenaltyDeposit(enrollment.utxo_key) == 0.coins)
```

Comment:
This function will return 0.coins if the utxo is spent or that stake was slashed. So even if the frozen UTXO is still in the pool, validator wont be able to reenroll with the same stake.